src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc

```diff
@@ -56,6 +56,10 @@ SourceGeneratorBase::~SourceGeneratorBase() {
 }
 
 void SourceGeneratorBase::WriteGeneratedCodeAttributes(io::Printer* printer) {
+  printer->Print("[global::System.Diagnostics.DebuggerNonUserCodeAttribute]\n");
+}
+
+void SourceGeneratorBase::WriteGeneratedTypeAttributes(io::Printer* printer) {
```

Comment:
It is a waste to have this empty method for future use only.
Let's add this when really needed.